core/pkg/hostsensorutils/hostsensordeploy.go

```diff
@@ -106,6 +132,11 @@ func (hsh *HostSensorHandler) applyYAML(ctx context.Context) error {
 			break
 		}
 	}
+	// check if namespace was already present on kubernetes
+	namespaceWasPresent, err = hsh.checkNamespaceWasPresent(namespaceName)
+	if err != nil {
+		return fmt.Errorf("failed to list namespaces: %v", err)
```

Comment:
```suggestion
		return err
```
[Comment]: Checking if the namespace is present is the logic of the inner function, so if you wish to include such details in the error message, you should write this in the function that originally raised the error.